isisd/isis_spf.c

```diff
@@ -1109,7 +1109,8 @@ static int isis_spf_process_lsp(struct isis_spftree *spftree,
 
 	if (fragnode) {
 		lsp = listgetdata(fragnode);
-		goto lspfragloop;
+		if (lsp->tlvs)
+			goto lspfragloop;
```

Comment:
While the fix for the crash looks good. But this will also mean that the processing will stop at the first fragment that does not have any fragment. What are the possibilities that a fragment has no TLVs in it. And then what are the possibilities that such a blank fragment is followed by a non-empty fragment in the fragment list.

Reply:
Also its not clear under which exact scenario did this crash ocurr?

Reply:
Does this crash happens on the same router where BGP routes are being redistributed into ISIS? Anyways stopping exabgp while injecting routes to for should not cause any LSP fragments to have zero TLVs. In my opinion along with this fix we need to fix the case where LSPs do not have any TLVs in it.

Reply:
> Does this crash happens on the same router where BGP routes are being redistributed into ISIS?

yes

> Anyways stopping exabgp while injecting routes to for should not cause any LSP fragments to have zero TLVs.

It is a exabgp issue. It means that this condition causes a denial of service

> In my opinion along with this fix we need to fix the case where LSPs do not have any TLVs in it.

My fix deals with this case

There is that at the beginning of the function. If no tlv, return. The previous patch was not so bad
```
	if (!lsp->tlvs)
		return ISIS_OK;
```

Reply:
What do you mean by exabgp issue? Is exabgp generating these empty ISIS LSP fragments? Not to my understanding. If these routes are learnt by FRR bgpd over a peering session with exabgp and then getting redistributed into ISIS, then this is still a ISISd LSP generation issue. If my assumptions are correct here, your fix is more like a hack and not the right fix.

Reply:
You are right. The issue is the generation of LSP.

Reply:
It appears with the introduction of classic LFA and when it is on. I have no time to investigate at the moment. I will try later